src/Microsoft.Data/DataFrameColumnCollection.cs

```diff
@@ -4,47 +4,54 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.ObjectModel;
 
 namespace Microsoft.Data
 {
     /// <summary>
     /// A DataFrameTable is just a container that holds a number of DataFrameColumns. It mainly acts as a convenient store to allow DataFrame to implement its algorithms
     /// </summary>
-    internal class DataFrameTable
+    public class DataFrameColumnCollection : Collection<DataFrameColumn>
     {
-        private IList<DataFrameColumn> _columns;
+        private Action ColumnsChanged;
 
         private List<string> _columnNames = new List<string>();
 
         private Dictionary<string, int> _columnNameToIndexDictionary = new Dictionary<string, int>(StringComparer.Ordinal);
 
         public long RowCount { get; internal set; }
 
-        public int ColumnCount { get; private set; }
-
-        public DataFrameTable()
+        public DataFrameColumnCollection(Action columnsChanged) : base()
```

Comment:
Do we want/need to have public constructors on this type? I kind of think it is only intended to be created from a `DataFrame`. So I don't think external callers should construct it.

Reply:
Nope. Made it internal.